### ISSUE TITLE
GRPOTrainer: logged reward/reward_std metrics ignore reward_weights

### ISSUE THREAD

**Issue by lailanelkoussy** [huggingface/trl#5352]
### System info

```
TRL main branch (commit on current HEAD)
Python 3.13, PyTorch 2.x, transformers latest
```

### Bug description

In `GRPOTrainer._generate_and_score_completions()`, the `reward` and `reward_std` metrics logged to training trackers (WandB, MLflow, TensorBoard) use an **unweighted** sum of per-function rewards, even when `reward_weights` are configured with non-uniform values.

The advantage computation correctly applies `self.reward_weights` (lines 1965 and 1997 in `grpo_trainer.py`), so models train correctly. However, the logging section at line 2022 overwrites the weighted `rewards` variable with `rewards_per_func.nansum(dim=1)` (unweighted) before logging:

```python
# Line 2022 — overwrites the correctly-weighted rewards variable:
rewards = rewards_per_func.nansum(dim=1)
self._metrics[mode]["reward"].append(rewards.mean().item())
self._metrics[mode]["reward_std"].append(rewards.std().item())
```

This means the dashboard `reward` metric diverges from the actual training signal whenever `reward_weights` are non-uniform.

### Reproduction

```python
# Two reward functions: one always returns 1.0, the other always returns 0.0
# With reward_weights=[0.7, 0.3]:
#   Expected logged reward: 0.7*1.0 + 0.3*0.0 = 0.7
#   Actual logged reward:   1.0 + 0.0 = 1.0

from trl import GRPOTrainer, GRPOConfig

config = GRPOConfig(
    output_dir="test_output",
    reward_weights=[0.7, 0.3],
    num_generations=4,
    report_to="none",
)

def reward_a(completions, **kwargs):
    return [1.0] * len(completions)

def reward_b(completions, **kwargs):
    return [0.0] * len(completions)

trainer = GRPOTrainer(
    model="Qwen/Qwen2.5-0.5B-Instruct",
    args=config,
    reward_funcs=[reward_a, reward_b],
    train_dataset=dataset,
)
trainer.train()
# trainer.state.log_history[-1]["reward"] == 1.0, should be 0.7
```

### Expected behavior

The logged `reward` metric should equal `mean(weighted_sum_per_sample)`, matching the actual training signal. With default weights (all 1.0), behavior is unchanged since weighted sum equals unweighted sum.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/huggingface/trl/issues) in the repo
